src/Features/Core/Portable/SolutionCrawler/SolutionCrawlerRegistrationService.cs

```diff
@@ -306,7 +307,8 @@ public Registration(int correlationId, Workspace workspace, SolutionCrawlerProgr
                 ProgressReporter = progressReporter;
             }
 
-            public Solution CurrentSolution => Workspace.CurrentSolution;
```

Comment:
Are we better off leaving the property named like this? It almost seems like the "just give me the solution" was abstracted here?

Reply:
Or even better "GetSolutionToAnalyze", because looking at the implementation I'm terrified of the number of places that the existing code was switching between Registration.CurrentSolution and Workspace.CurrentSolution; I could imagine we end up with code directly reaching to the ICompileTimeSolutionProvider and calling this method as well. A unique name might make bugs easier to spot?

Reply:
Will rename to GetSolutionToAnalyze.